### PR TITLE
Update docs for running test jobs in a cluster

### PR DESCRIPTION
**Pull request by pmorie**
Also remove a bind-mount flag onto the host for local binaries, which I didn't have and which took me a few min to figure out.

### CODE REVIEW

File: ci-operator/README.md [hunk 1]
```diff
@@ -320,32 +320,32 @@ available in the `registry.svc.ci.openshift.org/ci/test-infra:binaries` image
...
+`pull-ci-openshift-origin-master-e2e-aws` job against a pull request to
+`openshift/origin`. We'll specify the following pieces of information:
 
-The output of the script is a regular Pod that can be executed in the staging
-namespace in the [CI cluster](https://api.ci.openshift.org) and will report the
-results in the pull request, e.g.:
+* the repository: `openshift/origin`
+* the base branch of the repository to test: `master`
+* the pull request to merge onto the base branch: `21526`
+* the email address of github user that should have access to the namespace where the tests are run: `yourname@redhat.com`
```
Comment (stevekuznetsov) [inline ci-operator/README.md]: This should be the GitHub username

**Comment by stevekuznetsov** [comment]
Thanks @pmorie

File: ci-operator/README.md [hunk 2]
```diff
@@ -362,16 +362,26 @@ where you can determine the namespace where the `ci-operator` is running the job
 
 This will allow to determine the installer logs:
 
-```oc logs -f e2e -c setup  -n ci-op-3167h4v3 ```
+```
+oc logs -f e2e -c setup  -n ci-op-3167h4v3
+```
+
+Note, it is important to supply the name of your github account in order to be
+granted RBAC permission to view pods in the created test namespace.
```
Comment (bbguimaraes) [inline ci-operator/README.md]: And that the job passes `--give-pr-author-access-to-namespace` to `ci-operator` (all this assuming it is a `ci-operator` job).

File: hack/mkpjpod.sh [hunk 3]
```diff
@@ -20,7 +20,6 @@ run() {
     docker run \
         --rm \
         --volume "$PWD:/tmp/release:z" \
-        --volume ~/go/bin:/go/bin:z \
```
Comment (bbguimaraes) [inline hack/mkpjpod.sh]: *facepalm*

Thanks for catching this.

File: ci-operator/README.md (hunk 2 above)
Comment (pmorie) [reply ci-operator/README.md]: Do I need to update anything about this PR?

Comment (bbguimaraes) [reply ci-operator/README.md]: On second thought, all auto-generated jobs have that flag, so we can leave this out.

**Pull request merged**